tests/unit/test-firefox/test.firefox.js

```diff
@@ -263,6 +294,113 @@ describe('firefox', () => {
 
   });
 
+  describe('isDefaultProfile', () => {
+
+    it('detects common Firefox default profiles specified by name',
+       async () => {
+         const isDefault = await firefox.isDefaultProfile('default');
+         assert.equal(isDefault, true);
+
+         const isDevEditionDefault = await firefox.isDefaultProfile(
+           'dev-edition-default'
+         );
+         assert.equal(isDevEditionDefault, true);
+       });
+
+    it('allows profile name if is not listed as default in profiles.ini',
+      async () => {
+        return withTempDir(async (tmpDir) => {
+          const profilesDirPath = tmpDir.path();
+          const FakeProfileFinder = createFakeProfileFinder(profilesDirPath);
```

Comment:
we are actually overriding only `locateUserDirectory` static method (which is only used directly in our helper, because we manually pass it to the created `ProfileFinder` and so it will never be discovered again), all the other methods are the real ones.

I also remembered right now that I added the sinon spies to add some additional assertion... that I should add :-)